### PR TITLE
Pressure Changer and Coag & Floc Test Harness

### PR DESCRIPTION
**Pull request by MarcusHolly**
## Fixes/Resolves:

#1302 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/unit_models/tests/test_coag_floc_model.py [hunk 1]
```diff
@@ -9,585 +9,167 @@
...
+        ] = 0.009112749
+        self.unit_solutions[m.fs.unit.outlet.flow_mass_phase_comp[0, "Liq", "TDS"]] = (
+            0.01
         )
-        with pytest.raises(ConfigurationError, match=re.escape(error_msg)):
-            model.fs.properties = NaClParameterBlock()
-            model.fs.unit = CoagulationFlocculation(
-                property_package=model.fs.properties
-            )
-
```
Comment (adam-a-a) [inline watertap/unit_models/tests/test_coag_floc_model.py]: Where are we maintaining the checks on ConfigurationError and other expected exception handling? Is that part of the test harness?

**Comment by adam-a-a** [comment]
@lbianchi-lbl Maybe I am missing it, but I am not seeing the codecov checks here. Do codecov checks not run if the only changed files are test files? I wanted to check to see if test coverage decreased or not with this PR.

**Comment by lbianchi-lbl** [comment]
> @lbianchi-lbl Maybe I am missing it, but I am not seeing the codecov checks here. Do codecov checks not run if the only changed files are test files? I wanted to check to see if test coverage decreased or not with this PR.

@adam-a-a thanks for bringing this up.

- It looks like the Codecov step in the CI jobs is failing (which is consistent with the coverage reports not showing up as comments on this PR), but without causing the entire job to fail because the `fail_ci_if_error` option in `codecov/codecov-action` is not set to `true`
- I've restarted the two Python 3.8 jobs where the coverage report is generated and uploaded, which might fix the issue for this specific instance
- A more permanent fix requires a change to the CI configuration. I'm currently working on the same issue for IDAES so I might have something ready to use here as well

**EDIT** it looks like restarting the job didn't help. I'll try to have a PR up first thing tomorrow.

**Comment by hunterbarber** [comment]
I'd prefer to review this with the codecov for context (since largely this is minor change assuming the coverage is equivalent). Maybe @lbianchi-lbl can you just update the team when that gets resolved?

**Comment by lbianchi-lbl** [comment]
> I'd prefer to review this with the codecov for context (since largely this is minor change assuming the coverage is equivalent). Maybe @lbianchi-lbl can you just update the team when that gets resolved?

#1365 should address this, so feel free to leave a review to get it merged.

**Comment by lbianchi-lbl** [comment]
@adam-a-a @hunterbarber FYI: I realized I've made a mistake in the Codecov configuration in #1365, which causes the Codecov comment to erroneously report a 50% drop in coverage because the comment is created before all uploading jobs have run. This should be fixed by #1366.

**EDIT** the Codecov comment now seems to be working properly:

image

File: watertap/unit_models/tests/test_coag_floc_model.py (hunk 1 above)
Comment (hunterbarber) [reply watertap/unit_models/tests/test_coag_floc_model.py]: Yea, my comment was also going to be to re-add this in. [I just appended mine at the end of GAC after the harness](https://github.com/watertap-org/watertap/blob/main/watertap/unit_models/tests/test_gac.py#L312). Thoughts if that is the best implementation?

Comment (MarcusHolly) [reply watertap/unit_models/tests/test_coag_floc_model.py]: Seems like a good implementation to me. I've addressed this in the latest commit.

**Review by hunterbarber — APPROVED** [review]
LGTM after re-adding the logging coverage. Maybe we want to add a way to test error and warning logs in the test harness, but lets save that for when I also add the costing checks into the harness (not 1.0).

**Review by luohezhiming — APPROVED** [review]
LGTM

**Review by adam-a-a — APPROVED** [review]

**Pull request merged**